http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpGetSpec.scala

```diff
@@ -102,10 +101,30 @@ class HttpGetSpec
       val testGet = new StubbedHttpGet(Future.successful(response))
       testGet.GET[HttpResponse](url, Seq("header" -> "foo")).futureValue shouldBe response
     }
+
     "be able to return objects deserialised from JSON" in {
       val testGet = new StubbedHttpGet(Future.successful(HttpResponse(200, """{"foo":"t","bar":10}""")))
       testGet.GET[TestClass](url, Seq("header" -> "foo")).futureValue should be(TestClass("t", 10))
     }
+
+    "be able to return Some[T] when deserialising as an option of object from JSON" in {
+      val testGet = new StubbedHttpGet(Future.successful(HttpResponse(200, """{"foo":"t","bar":10}""")))
```

Comment:
Is this one of the tests from `http-verbs-example`? I think the style in the HeadersSpec serves as a better example of how to use the library - i.e. creating a client.
Maybe we need a separate class to demonstrate how to use the library, separate from us testing that the library behaves as expected. The intention of the tests will be clear when maintaining the library. And we can create a link from the README to tell users to look at those tests.

Reply:
Have added Examples as separate tests in the project and linked from the README.